### PR TITLE
Don't report unused parameter for GetInstance of ICustomMarshaler

### LINKED ISSUE #45743: IDE0060 incorrectly reported for the GetInstance method of ICustomMarshaler implementation
**Version Used**: 

**Steps to Reproduce**:

1. Implement a custom marshaler (`ICustomMarshaler`)
2. Define the [`GetInstance` method](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.icustommarshaler?view=netcore-3.1#implementing-the-getinstance-method) of the custom marshaler, but don't use the string parameter

**Expected Behavior**:

IDE0060 is not reported.

**Actual Behavior**:

IDE0060 is reported, saying the required parameter can be removed.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #45743

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs [hunk 1]
```diff
@@ -1580,6 +1580,38 @@ public async Task RecordPrimaryConstructorParameter_PublicRecord()
...
+    public int GetNativeDataSize()
+        => throw new NotImplementedException();
+
+    public IntPtr MarshalManagedToNative(object ManagedObj)
+        => throw new NotImplementedException();
+
+    public object MarshalNativeToManaged(IntPtr pNativeData)
+        => throw new NotImplementedException();
+        
+    public ICustomMarshaler GetInstance(string [|s|])
```
Comment (Youssef1313) [inline src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedParametersTests.cs:1612]:
```suggestion
    public static ICustomMarshaler GetInstance(string [|s|])
```

**Comment by Youssef1313** [comment]
@sharwell @CyrusNajmabadi Please review.

File: src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -291,6 +291,17 @@ public bool IsComputingUnusedParams(ISymbol symbol)
                 return !symbol.HasPublicResultantVisibility();
             }
 
+            // Don't report on valid GetInstance method of ICustomMarshaler.
+            // See https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.icustommarshaler#implementing-the-getinstance-method
+            if (symbol is IMethodSymbol methodSymbol && methodSymbol.MetadataName == "GetInstance" && methodSymbol.IsStatic &&
+                methodSymbol.Parameters.Length == 1 && methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
+                !methodSymbol.ContainingType.Interfaces.IsEmpty && methodSymbol.ContainingType.Interfaces[0].Name == "ICustomMarshaler")
```
Comment (Youssef1313) [inline src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs:298]: What a bad code!
Will re-work it :smile:

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Can you review this small fix? Thanks.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**